extensions/eclipse-che-theia-remote-impl-k8s/src/node/k8s-plugin-service-impl.ts

```diff
@@ -0,0 +1,739 @@
+      throw new Error('Plugin registry URL is not configured.');
+    }
+
+    if (!registryInternalURL) {
+      // console.error('Plugin registry internal URL is not configured.');
+      throw new Error('Plugin registry internal URL is not configured.');
+    }
+
+    this.defaultRegistry = {
+      name: 'Eclipse Che plugins',
+      internalURI: this.trimTrailingSlash(registryInternalURL),
+      publicURI: this.trimTrailingSlash(registryURL),
+    };
+
+    return this.defaultRegistry;
+  }
+
+  async getInstalledPlugins(): Promise<string[]> {
+    await this.deferred.promise;
+
+    try {
+      const list = await this.getPluginList();
+      if (list.plugins) {
+        const installed: string[] = [];
+
+        for (let plugin of list.plugins) {
+          const parts: string[] = plugin.split('.');
+          plugin = `${parts[0]}/${parts[1]}/latest`;
+          installed.push(plugin);
+        }
+
+        return installed;
+      }
+    } catch (error) {
+      throw new Error('Unable to get list of installed plugins. ' + error.message);
+    }
+
+    return [];
+  }
+
+  async installPlugin(pluginKey: string): Promise<void> {
+    await this.deferred.promise;
+
+    try {
+      const keys: string[] = pluginKey.split('/');
+
+      const publisher = keys[0];
+      const name = keys[1];
+      const version = keys.length === 3 ? keys[2] : 'latest';
+
+      const pluginID = `${publisher}/${name}/${version}`;
+
+      // check whether the plugin is already installed
+      const installedPlugins = await this.getInstalledPlugins();
+      if (installedPlugins.find(plugin => plugin === pluginID)) {
+        return;
+      }
+
+      const yaml = await this.getPluginMetaYaml(pluginID);
+
+      if (!yaml.spec) {
+        const msg = `Failure to read .spec section.`;
+        // console.error(msg);
+        throw new Error(msg);
+      }
+
+      // check, whether the sidecar container is specified
+      if (yaml.spec.containers) {
+        // add plugin to sidecar container / universal developer image
+        await this.installPluginToDeveloperContainer(yaml);
+      } else {
+        // add plugin to theia-ide container
+        await this.installPluginToEditorContainer(yaml);
+      }
+
+      // persist plugin ID to /plugins/che-plugins.json file
+      await this.addPluginToChePluginsJSON(`${publisher}.${name}`);
+    } catch (error) {
+      const msg = `Unable to install plugin ${pluginKey}. ${error.message ? error.message : error}`;
+      // console.error(msg);
+      throw new Error(msg);
+    }
+  }
+
+  /**
+   * Fetches plugin meta.yaml
+   * @param plugin pluin in format publisher/name/version
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  async getPluginMetaYaml(plugin: string): Promise<any> {
+    const registryURL = this.trimTrailingSlash(this.env.getPluginRegistryInternalURL());
+    const yamlURL = `${registryURL}/plugins/${plugin}/meta.yaml`;
```

Comment:
with DevWorkspaces we should not read meta.yaml

we read the files che-theia-plugin.yaml
example:
https://eclipse-che.github.io/che-plugin-registry/main/v3/plugins/redhat/java/latest/che-theia-plugin.yaml

all meta.yaml will be dropped after the switch to DevWorkspaces

Reply:
Done